cubical/data_handler/ms_tile.py

```diff
@@ -147,34 +147,42 @@ def load_montblanc_models(self, uvwco, loaded_models, model_source, cluster, imo
                 ddid_index, uniq_ddids, _ = data_handler.uniquify(self.ddid_col)
 
                 self._freqs = np.array([self.tile.dh.chanfreqs[ddid] for ddid in uniq_ddids])
-
-                self._row_identifiers = ddid_index * n_bl * ntime + (self.times - self.times[0]) * n_bl + \
+                
+                def timestep_index(times, tol=1.0e-9):
+                    """ Compute the prescan operation to find the unique timestep idenfiers for
+                        a TIME_CENTROID array """
+                    tindx = np.zeros_like(times, dtype=np.int64)
+                    tindx[1:] = np.abs(times[1:] - times[:-1]) > tol
+                    tindx = np.add.accumulate(tindx)
+                    return tindx
+
+                self._row_identifiers = ddid_index * n_bl * ntime + timestep_index(self.times) * n_bl + \
```

Comment:
I am not convinced by this - in my opinion we are trying to fix something which should not be broken. We need to find the root cause otherwise we may have unexpected behaviour elsewhere. Doing a search for times only yields 25 results and I cannot see any reason for self.times to become non-integer. For reference, self.time_col contains float values from the TIME column of the MS. self.times is of the same shape but instead contains integer indices associating each time with the timeslot/integration to which it belongs. These indices are generated by uniquify, which is defined __init__.py in the data_handler folder.

Reply:
It must be broken upstream from this method somewhere. I can revert the change and if you run it with py3 it will break

Reply:
I think the safest thing is to sprinkle asserts for the data types into the codebase as they can always be disabled with python if you want full performance

Reply:
Yeah there's no way ``times`` should be float. I vote to rename it though, as from the variable name it's not obvious at all that this means "timeslot index" (anyone new to the code base will just take it to mean the plural of "time").

Reply:
Ok I found the difference it was stemming from your n_bl calculation which I already fixed with //

The relative difference is still
```
*** mean relative diff between CORRECTED_DATA and DE_DATA is -35.98283290863037 dB
*** ninety fifth percentile relative diff between CORRECTED_DATA and DE_DATA is -31.854880104464907 dB
```
which points to this being the sin projection you previously employed